minimalkv/contrib/__init__.py

```diff
@@ -20,14 +20,15 @@ class ExtendedKeyspaceMixin:
     Note: This Mixin is unsupported and might not work correctly with all backends.
     """
 
-    def _check_valid_key(self, key: Optional[str]):
+    def _check_valid_key(self, key: Optional[str]) -> None:
         """Checks if a key is valid and raises a ValueError if its not.
 
         When in need of checking a key for validity, always use this
         method if possible.
 
         :param key: The key to be checked
         """
+        # TODO: Is there any case where key is None?
```

Comment:
Yes, it in the case of at least the Redis store. I think there you can pass `None` on `put` and get a new (random) key as a result.

Reply:
In that case we should overwrite the method in the Redis store any not here IMO.